src/components/Calendar/MucCalendar.vue

```diff
@@ -0,0 +1,452 @@
+ * @param newValue is the newly selected date.
+ */
+const updateMVRange = (newValue: Date) => {
+  if (selectedDate.value === null || Array.isArray(selectedDate.value))
+    selectedDate.value = { from: null, to: null };
+  else if (selectedDate.value instanceof Date)
+    selectedDate.value = { from: selectedDate.value, to: null };
+
+  selectedDate.value =
+    !selectedDate.value.from || selectedDate.value.to
+      ? { from: newValue, to: null }
+      : isDateAfterOther(selectedDate.value.from, newValue)
+        ? { from: newValue, to: selectedDate.value.from }
+        : { from: selectedDate.value.from, to: newValue };
+};
+
+/**
+ * Handles a date click event based on the selected variant.
+ * Depending on whether the variant is "single", "multiple", or "range", it calls the appropriate function to update the selected date(s).
+ * @param date - The date that was clicked.
+ */
+const clickedDate = (date: Date) => {
+  if (props.disabled) return;
+  switch (props.variant) {
+    case "single":
+      updateMVSingle(date);
+      break;
+    case "multiple":
+      updateMVMultiple(date);
+      break;
+    case "range":
+      updateMVRange(date);
+      break;
+  }
+};
+
+/**
+ * Clicked on a broader selection (month or year) to be set for the more detailed view.
+ * Changes view afterward.
+ * @param date date within in the clicked month or year
+ */
+const clickedBroadSelection = (date: Date) => {
+  viewTransition.value = "view-detail";
+  viewDate.value = new Date(date.getFullYear(), date.getMonth());
+  detailedView();
+};
+
+/**
+ * Changes view to a broader view-type (month or year)
+ */
+const broaderView = () => {
+  viewTransition.value = "view-broad";
+  if (view.value === "day") view.value = "month";
+  else if (view.value === "month") view.value = "year";
+};
+
+/**
+ * Changes view to a detailed view-type (day or month)
+ */
+const detailedView = () => {
+  if (view.value === "year") view.value = "month";
+  else if (view.value === "month") view.value = "day";
+};
+
+/**
+ * Providing data for the day selection view - reduces amount of props
+ */
+provide(MucCalendarKey, {
+  viewDate,
+  selectedDate,
+  variant: readonly(toRef(props.variant)),
+  showAdjacentMonths: readonly(toRef(props.showAdjacentMonths)),
+});
+</script>
+
+<style scoped>
+/*animation - backwards is not working here because of interchanging of the animations*/
+
+@keyframes appear-broad {
+  0% {
+    transform: scale(1.7);
+    opacity: 0;
+  }
+  100% {
+    opacity: 1;
+    transform: scale(1);
+  }
+}
+
+@keyframes disappear-broad {
+  0% {
+    transform: scale(1);
+    opacity: 1;
+  }
+  100% {
+    opacity: 0;
+
+    transform: scale(0.7);
+  }
+}
+
+@keyframes appear-detail {
+  0% {
+    transform: scale(0.7);
+    opacity: 0;
+  }
+  100% {
+    opacity: 1;
+    transform: scale(1);
+  }
+}
+
+@keyframes disappear-detail {
+  0% {
+    transform: scale(1);
+    opacity: 1;
+  }
+  100% {
+    opacity: 0;
+
+    transform: scale(1.7);
+  }
+}
```

Comment:
Thought about that - but all the animations etc belong very closely to this component... If we separate this we technically would need to do this for a lot more component.
Otherwise this has the benefit that the css classes are only accessible for the calendar components

Reply:
Yes I thought about that, too. But what you could do is create a custom css which resides `src/components/Calendar/` and is imported only here. It makes the components code way easier to read.